ceno_zkvm/examples/riscv_opcodes.rs

```diff
@@ -112,38 +114,49 @@ fn main() {
     let prog_config = zkvm_cs.register_table_circuit::<ExampleProgramTableCircuit<E>>();
     zkvm_cs.register_global_state::<GlobalState>();
 
-    for instance_num_vars in args.start..args.end {
-        let step_loop = 1 << (instance_num_vars - 1); // 1 step in loop contribute to 2 add instance
+    let mut zkvm_fixed_traces = ZKVMFixedTraces::default();
 
-        // init vm.x1 = 1, vm.x2 = -1, vm.x3 = step_loop
-        let program_data: &[u32] = &[1, u32::MAX, step_loop];
+    zkvm_fixed_traces.register_table_circuit::<ExampleProgramTableCircuit<E>>(
+        &zkvm_cs,
+        &prog_config,
+        &program,
+    );
 
-        let mut zkvm_fixed_traces = ZKVMFixedTraces::default();
+    let reg_init = initial_registers();
+    // Define program constant here
+    let program_data: &[u32] = &[];
+    let program_data_init = init_program_data(program_data);
 
-        zkvm_fixed_traces.register_table_circuit::<ExampleProgramTableCircuit<E>>(
-            &zkvm_cs,
-            &prog_config,
-            &program,
-        );
+    config.generate_fixed_traces(
+        &zkvm_cs,
+        &mut zkvm_fixed_traces,
+        &reg_init,
+        &program_data_init,
+    );
 
-        let reg_init = initial_registers();
-        let mem_init = initial_memory(program_data);
+    let pk = zkvm_cs
+        .clone()
+        .key_gen::<Pcs>(pp.clone(), vp.clone(), zkvm_fixed_traces)
+        .expect("keygen failed");
+    let vk = pk.get_vk();
 
-        config.generate_fixed_traces(&zkvm_cs, &mut zkvm_fixed_traces, &reg_init, &mem_init);
+    // proving
+    let prover = ZKVMProver::new(pk);
+    let verifier = ZKVMVerifier::new(vk);
 
-        let pk = zkvm_cs
-            .clone()
-            .key_gen::<Pcs>(pp.clone(), vp.clone(), zkvm_fixed_traces)
-            .expect("keygen failed");
-        let vk = pk.get_vk();
+    for instance_num_vars in args.start..args.end {
+        // The performance benchmark is hook on number of "add" opcode instances.
+        // Each iteration in the loop contributes 2 add instances,
+        // so we divide by 2 here to ensure "instance_num_vars" aligns with the actual number of add instances.
```

Comment:
@hero78119 

What does this mean?

What does `instance_num_vars` mean?

Reply:
It means the number of variables in the multilinear polynomial which encodes witnesses of the add opcode circuit. For example, if we have 2^10 add execution traces, then we can use ml polynomials of nv = 10 to encode the witnesses (`rs1.value: UInt, rs2.value: UInt, rd.value: UInt`).